packages/clarity-decode/src/data.ts

```diff
@@ -71,8 +71,19 @@ export function decode(tokens: Data.Token[]): DataEvent {
             let e = 2; // Start from 3rd index since first two are used for time & event
             let extract: Data.ExtractData = {};
             while (e < tokens.length) {
-                extract[tokens[e++] as string | number] = tokens[e++] as string;
+                if (typeof(tokens[e + 1]) == Constant.String) {
```

Comment:
Assuming this is for backward compatibility. If so, could you please add a comment with version number so it's easy to track in future.